dd-sdk-android/src/main/kotlin/com/datadog/android/rum/internal/RumFeature.kt

```diff
@@ -117,26 +123,36 @@ internal object RumFeature : SdkFeature<RumEvent, Configuration.Feature.RUM>() {
 
     private fun initializeVitalMonitors() {
         vitalExecutorService = ScheduledThreadPoolExecutor(1)
-        val cpuReaderRunnable = VitalReaderRunnable(
-            CPUVitalReader(),
-            cpuVitalMonitor,
-            vitalExecutorService,
-            VITAL_UPDATE_PERIOD_MS
-        )
-        vitalExecutorService.schedule(
-            cpuReaderRunnable,
-            VITAL_UPDATE_PERIOD_MS,
-            TimeUnit.MILLISECONDS
-        )
 
-        val memoryReaderRunnable = VitalReaderRunnable(
-            MemoryVitalReader(),
-            memoryVitalMonitor,
+        initializeVitalMonitor(CPUVitalReader(), cpuVitalMonitor)
+        initializeVitalMonitor(MemoryVitalReader(), memoryVitalMonitor)
+
+        val frameRateReader = FrameRateVitalReader()
+        initializeVitalMonitor(frameRateReader, frameRateVitalMonitor)
+        try {
```

Comment:
The looper on which we are running is mostly to know on which thread the callback will be delivered, the framerate reported is independent from any looper and linked with the app's rendering

Reply:
Ok, got it. If we are not on the thread which has looper, maybe we could spin our own thread with `Looper.prepare` + `Looper.loop` and listen there? wdyt? Obviously we cannot do it on the original thread, because it will be blocked forever by `Looper.loop`, which is not good thing for threads from the pool, like if SDK initialized using RxJava worker thread, or with co-routines. Also I'm wondering if it is a common case to init SDK from background thread (which in 99.9% cases won't have looper attached).

Reply:
I didn't find any recommendations to initialiaze SDK on the main thread in the docs. I'm pretty sure that big apps (and which result in big number of sessions probably) which optimize startup time will put 3rd party libs init on the background thread. And if it is the case (init on background thread) we can still push `Choreographer` registration to the main thread by doing something like `Handler(Looper.getMainLooper()).post { register choreographer }` (btw we already use main thread `Handler` in the `DatadogRumMonitor`), so that such users still have the metric collected.

Anyway, this is just my opinion, I won't push for it :)